ItemPool.py

```diff
@@ -961,6 +957,13 @@ def get_pool_core(world: World) -> tuple[list[str], dict[str, Item]]:
 
     for item, maximum in item_difficulty_max[world.settings.item_pool_value].items():
         replace_max_item(pool, item, maximum)
+    # Dynamically condense regular heart pieces into heart containers depending on how many are in the pool
+    # (which varies based on the Shuffle Gerudo Fortress Heart Piece setting)
+    if world.settings.item_pool_value in ('plentiful', 'ludicrous'):
+        indices = [items_idx for items_idx, val in enumerate(pool) if val == 'Piece of Heart']
+        num_full_hearts = (len(indices) // 4) * 4
+        for hearts_idx, items_idx in enumerate(indices[:num_full_hearts]):
+            pool[items_idx] = 'Heart Container' if hearts_idx % 4 == 0 else get_junk_item()[0]
```

Comment:
Has this been tested?

Reply:
Yes. Here's an overview of the generated item pools:

|Shuffle Gerudo Fortress Heart Piece|Item Pool|Heart Container|Piece of Heart|Piece of Heart (Treasure Chest Game)|Piece of Heart (Out of Logic)|
|---|---|--:|--:|--:|--:|
|(no hideout ER)|Plentiful|16|3|1|0|
|(no hideout ER)|Ludicrous|16|3|1|0|
|Remove|Plentiful|16|3|1|0|
|Remove|Ludicrous|16|3|1|0|
|Vanilla|Plentiful|16|3|1|0|
|Vanilla|Ludicrous|16|3|1|0|
|Shuffle|Plentiful|17|0|1|0|
|Shuffle|Ludicrous|17|0|1|0|

This matches my expectations. (`Piece of Heart (Out of Logic)` is not listed as part of the item pool because it's not shuffled.)